phpBB/phpbb/messenger/queue.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\messenger;
+
+use phpbb\config\config;
+use phpbb\event\dispatcher;
+use phpbb\di\service_collection;
+use phpbb\filesystem\filesystem;
+
+/**
+ * Handling messenger file queue
+ */
+class queue
+{
+	/** @var string */
+	protected $cache_file;
+
+	/** @var config */
+	protected $config;
+
+	/** @var array */
+	protected $data = [];
+
+	/** @var dispatcher */
+	protected $dispatcher;
+
+	/** @var \phpbb\filesystem\filesystem_interface */
+	protected $filesystem;
+
+	/** @var service_collection */
+	protected $messenger_method_collection;
+
+	/** @var int */
+	protected $package_size = 0;
```

Comment:
Seems unused?

Reply:
It's used in `phpbb_email` class in `queue->init()` call.